### PR TITLE
Add latex styles for a more functional manual.

### PR DESCRIPTION
**Pull request by bangerth**
Specifically, use better defaults for the 'listings' and 'hyperref'
styles.

### CODE REVIEW

**Comment by ljhwang** [comment]
We have been working on a latex template for manuals.  This should appear in the geodynamics Best Practices repo shortly.  This is a near relative of the ASPECT manual so i suspect the style follows closely your proposal.

File: doc/src/user_guide/Rayleigh_User_Guide.tex [hunk 1]
```diff
@@ -31,6 +30,33 @@
   breakatwhitespace=true,
   tabsize=3
 }
+\lstset{
+  language=C++,
```
Comment (gassmoeller) [inline doc/src/user_guide/Rayleigh_User_Guide.tex]: Is there a fortran option? Or does it not matter, which language is actually selected?

File: doc/src/user_guide/Rayleigh_User_Guide.tex [hunk 2]
```diff
@@ -31,6 +30,33 @@
...
+  showstringspaces=false,
+  basicstyle=\small\ttfamily,
+  columns=fullflexible,
+  keepspaces=true,
+  frame=single,
+  breaklines=true,
+  postbreak=\raisebox{0ex}[0ex][0ex]{\hspace{5em}\ensuremath{\color{red}\hookrightarrow\space}}
+}
+\lstdefinelanguage{prmfile}{morekeywords={set,subsection,end},
+                            morecomment=[l]{\#},escapeinside={\%\%}{\%},}
```
Comment (gassmoeller) [inline doc/src/user_guide/Rayleigh_User_Guide.tex]: I do not think we need the prmfile language here. We could still keep it and modify it for the Rayleigh inputfile later. What is your opinion?

File: doc/src/user_guide/Rayleigh_User_Guide.tex [hunk 3]
```diff
@@ -31,6 +30,33 @@
...
+\lstdefinelanguage{gnuplot}{morekeywords={plot,using,title,with,set,replot},
+                            morecomment=[l]{\#},}
+
+
+% use the hyperref package; set the base for relative links to
+% the top-level aspect directory so that we can link to
```
Comment (gassmoeller) [inline doc/src/user_guide/Rayleigh_User_Guide.tex]: aspect -> Rayleigh

File: doc/src/user_guide/Rayleigh_User_Guide.tex [hunk 4]
```diff
@@ -31,6 +30,33 @@
...
+% files in the aspect tree without having to specify the
```
Comment (gassmoeller) [inline doc/src/user_guide/Rayleigh_User_Guide.tex]: aspect -> Rayleigh

File: doc/src/user_guide/Rayleigh_User_Guide.tex (hunk 1 above)
Comment (bangerth) [reply doc/src/user_guide/Rayleigh_User_Guide.tex]: I don't think any is selected by default. This block just defines the C++ style -- which we don't need here, but I thought I'd add it as an example for how one would define another language style. The f90 style is already defined a few lines above.

This isn't important, so I removed the block.

File: doc/src/user_guide/Rayleigh_User_Guide.tex (hunk 2 above)
Comment (bangerth) [reply doc/src/user_guide/Rayleigh_User_Guide.tex]: Yes, that was the idea.

**Comment by bangerth** [comment]
All fixed.

**Pull request merged**